lib/ansible/module_utils/facts/virtual/linux.py

```diff
@@ -220,6 +220,17 @@ def get_virtual_facts(self):
                 virtual_facts['virtualization_role'] = 'guest'
                 return virtual_facts
 
+        # In older Linux Kernel versions, /sys filesystem is not available
+        # dmidecode is the safest option to parse virtualization related values
+        dmi_bin = self.module.get_bin_path('dmidecode')
+        (rc, out, err) = self.module.run_command('%s -s system-product-name' % dmi_bin)
+        if rc == 0:
+            # Strip out commented lines (specific dmidecode output)
+            vendor_name = ''.join([line.strip() for line in out.splitlines() if not line.startswith('#')])
+            if vendor_name in ['VMware Virtual Platform', 'VMware7,1']:
+                virtual_facts['virtualization_type'] = 'VMware'
+                virtual_facts['virtualization_role'] = 'guest'
```

Comment:
@Akasurde @alikins here is a ```return virtual_facts``` missing.
Otherwise the facts are reset to 'NA' later!

Reply:
Better just fix tiny things right away before they get released ;).